src/Invoice.php

```diff
@@ -609,15 +641,29 @@ protected function formatAmount($amount)
         return Cashier::formatAmount($amount, $this->invoice->currency);
     }
 
+    /**
+     * Get the tax rate from tax rate details, fetching from Stripe if needed.
+     *
+     * @param  object  $taxRateDetails
+     * @return \Stripe\TaxRate|null
+     */
+    protected function getTaxRate($taxRateDetails)
+    {
+        // If tax_rate is already expanded as an object, return it
+        if (isset($taxRateDetails->tax_rate) && is_object($taxRateDetails->tax_rate) && isset($taxRateDetails->tax_rate->id)) {
+            return $taxRateDetails->tax_rate;
+        }
+
+        return null;
```

Comment:
Shouldn't this check if `$taxRateDetails->tax_rate` is an instance of `Stripe\TaxRate`?

```suggestion
        return $taxRateDetails->tax_rate instanceof \Stripe\TaxRate ? $taxRateDetails->tax_rate : null;
```